python/rmm/_lib/memory_resource.pyx

```diff
@@ -26,9 +26,19 @@ from libcpp.string cimport string
 from cuda.cudart import cudaError_t
 
 from rmm._cuda.gpu import CUDARuntimeError, getDevice, setDevice
+
 from rmm._lib.cuda_stream_view cimport cuda_stream_view
 
 
+# NOTE: We can't use declarations of CUDA types from CUDA Python and
+# expect them to work with external code. So we declare them
+# ourselves.  See https://github.com/NVIDIA/cuda-python/issues/22 for
+# details.
+cdef extern from "driver_types.h" nogil:
```

Comment:
Problem: cudaMemAllocationHandleType doesn't exist in CUDA < 11.2.

Reply:
Should this be using the same kind of tricks as in https://github.com/rapidsai/rmm/pull/990? cc: @jakirkham

Reply:
cc @robertmaynard

Reply:
As @shwina  stated we want a shim type. My initial though is that the constructor shouldn't take a
`thrust::optional<cudaMemAllocationHandleType>` but a `thrust::optional<rmm::cudaMemAllocationHandleType>` which we map to the correct cuda driver type. 

The rmm type would always exist so we can have a single constructor signature that works no matter the cuda toolkit version

Reply:
Yeah that's a better idea. Another option would be to not let the user specify the handle type, and only support Posix handles on the C++ side. That might be overly limiting.